src/components/TotalPostsContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import { getTotalPosts, writing } from '../api/api';
+import Post from './Post';
+import { AxiosError } from 'axios';
+export const URL = `page=1&pageSize=5`;
+const TotalPostsContainer = () => {
+  const router = useRouter();
+  const { orderBy, keyword } = router.query;
+
+  const [posts, setPosts] = useState<writing[]>([]);
+  const getPosts = async () => {
+    if (orderBy) {
+      try {
+        const result = await getTotalPosts(`${URL}&orderBy=${orderBy}`);
+        setPosts(result);
+      } catch (error) {
+        const err = error as AxiosError;
+      }
+    }
+    if (keyword) {
+      try {
+        const result = await getTotalPosts(`${URL}&keyword=${keyword}`);
+        setPosts(result);
+      } catch (error) {
+        const err = error as AxiosError;
+      }
+    }
+  };
+  useEffect(() => {
+    getPosts();
+  }, [orderBy, keyword]);
```

Comment:
```suggestion
  const getPosts = async () => {
 
      try {
        const result = await getTotalPosts(`${URL}&orderBy=${orderBy}`);
        setPosts(result);
      } catch (error) {
        const err = error as AxiosError;
      }
    }
    if (keyword) {
      try {
        const result = await getTotalPosts(`${URL}&keyword=${keyword}`);
        setPosts(result);
      } catch (error) {
        const err = error as AxiosError;
      }
    
  };
  useEffect(() => {
     if (orderBy) {
    getPosts();
    }
  }, [orderBy, keyword]);
```
요렇게 작성하는건 어떨까요?  useEffect구문인데 sortBy가 있을때만 해당 함수가 실행되게요! 위처럼 작성하면 함수가 sortBy에 종속돼서 재사용이 힘들것 같아요!

Reply:
@endmoseung  멘토님 keyword가 검색 인풋의 string 값이고 orderby는 select태그의 정렬 방식입니다. 기본적으로 정렬값이 default로 되어있어서 useEffect의 2개의 의존성배열을 사용 하였습니다.